AK/FloatingPointStringConversions.cpp

```diff
@@ -1711,21 +1743,15 @@ class MinimalBigInt {
         add_small(value);
     }
 
-    void append(u64 word)
-    {
-        VERIFY(m_used_length <= max_words_needed);
-        m_words[m_used_length] = word;
-        ++m_used_length;
-    }
-
     // The max valid words we might need are log2(10^(769 + 342)), max digits + max exponent
-    static constexpr size_t max_words_needed = 58;
+    static constexpr size_t words_in_u64 = word_size == 64 ? 1 : 2;
+    static constexpr size_t max_words_needed = 58 * words_in_u64;
 
     size_t m_used_length = 0;
 
     // FIXME: This is an array just to avoid allocations, but the max size is only needed for
     //        massive amount of digits, so a smaller vector would work for most cases.
-    Array<u64, max_words_needed> m_words {};
+    Array<NativeWord, max_words_needed> m_words {};
```

Comment:
errrr… why do we care for `u64` in `max_words_needed` if weʼre using native word here? 🤔

Reply:
As comment a few lines above state, we need at most $\log_2(10^{769 + 342})$ bits, so it is 116 words for `NativeWord == u32` and 58 words for `NativeWord == u64`.

Reply:
i believe that `NativeWord` canʼt be `u32`…? 🤔

Reply:
(answered in other comment — please resolve this)